### DISCUSSION TITLE
--limit-max-requests does not reload Workers if only Websocket Client connected

### DISCUSSION THREAD

**Discussion by lesc6** [Kludex/uvicorn#2146] [Potential Issue]
how reproduce Issue: 
`uvicorn --host=0.0.0.0 --port=8000 --limit-max-requests=5 --workers=1 `
Use any Websocket Client:
  - websockets
  - wscat 
  
you expect that Worker process will be killed after N WebSocket connection, but it will never killed  and it consume more and more memory.

as far as I understand, when a WebSocket connection is established, it does not consider it as request.

**Comment by lesc6** [comment]
https://github.com/benoitc/gunicorn/issues/3086

**Comment by M1ha-Shvn** [comment]
This is a real problem for WebSocket handlers without http requests as there is no any built-in ability to reload workers and clean memory. I understand, this does not work with `--limit-max-requests` flag, but it would be very suitable to create such a flag for websockets. For instance, to softly-reboot worker after closing N websocket connections. In my case, workers after some time of work consume 4-5 times more memory than my app really uses (according to [memray](https://bloomberg.github.io/memray/getting_started.html) cache metric). After worker reboot this difference is gone.
